docs/timeseries_pipeline_elements.md

```diff
@@ -0,0 +1,367 @@
+# Timeseries Pipelines [<sup><mark>experimental</mark></sup>](/docs/README.md#tag-notes)
+
+> [Description](#timeseries-pipeline-description)<br>
+> [Example](#timeseries-pipeline-example)<br>
+> [Pipeline Elements](#timeseries-pipeline-elements)
+
+## Description <a id="timeseries-pipeline-description"></a>
+
+Timescale timeseries objects are just a convenient and efficient way of tracking a single value over time and are detailed a bit more [here](timeseries.md).  One of our primary goals with timeseries is that they should be easy and efficient to perform basic operations on, and that is where pipelines enter the picture.  At its simplest, a pipeline is just a timeseries connected to a [pipeline element](#timeseries-pipeline-elements) via the pipeline operator `|>`.  However, most pipeline operations output new timeseries, so it's possible to chain many pipeline elements together such that the output from one element become the input to the next.
+
+### A note on operator associativity and grouping
+
+Due to limitations in the PostgresQL parser, custom operators are required to be left associative.  The following pipeline will always result in `elementA` being applied to `timeseries` and then `elementB` being applied to the result.
+
+```SQL ,ignore
+SELECT timeseries |> elementA |> elementB;
+```
+
+However, it is possible to explicitly group elements using parentheses:
+
+```SQL ,ignore
+SELECT timeseries |> (elementA |> elementB);
+```
+
+This will result in a pipeline object being created from elements A and B, which will then be applied to the timeseries.  While we don't presently take maximum advantage of this internally, these multiple element pipelines should enable optimizations moving forward.  Therefore, this second form should be preferred where possible.
+
+## Usage Example <a id="timeseries-pipeline-example"></a>
+
+For this example let start with a table of temperatures collected from different devices at different times.
+
+```SQL ,non-transactional,ignore-output
+SET TIME ZONE 'UTC';
+CREATE TABLE test_data(time TIMESTAMPTZ, device INTEGER, temperature DOUBLE PRECISION);
+```
+
+In order to have some nominally interesting data to look at, let's populate this table with random data covering 30 days of readings over 10 devices.
+
+```SQL ,non-transactional,ignore-output
+SELECT setseed(0.456);
+INSERT INTO test_data 
+    SELECT
+        '2020-01-01 00:00:00+00'::timestamptz + ((random() * 2592000)::int * '1 second'::interval),
+        floor(random() * 10 + 1),
+        50 + random() * 20
+    FROM generate_series(1,10000);
+```
+
+Now suppose we want to know how much the temperature fluctuates on a daily basis for each device.  It is actually simple to do this using timeseries and pipelines:
```

Comment:
```suggestion
Now suppose we want to know how much the temperature fluctuates on a daily basis for each device.  Using timeseries and pipelines can simplify the process of finding the answer:
```
(not fully happy with the suggestion, but we generally try to avoid saying things are "easy" and "simple". It's not to everyone. ;-)